src/group_theory/subgroup.lean

```diff
@@ -1375,3 +1376,64 @@ instance : is_modular_lattice (subgroup C) :=
 end⟩
 
 end subgroup
+
+section pointwise
+
+namespace subgroup
+
+@[to_additive]
+lemma sup_eq_closure (H K : subgroup G) : H ⊔ K = closure (H * K) :=
```

Comment:
Can you prove the statement
```
lemma closure_sup_closure (H K : set G) : closure H ⊔ closure K = closure (H * K)
```
If so, then your statement falls out from rewriting `closure_eq` on the LHS

Reply:
I don't think that's a true statment. For instance, in `ℤ` one has `closure {1} ⊔ closure {1} ≠ closure ({1} + {1})`.

Reply:
Ah, good point. It's true as a `≤` though right?

Reply:
Yes, that's correct. I added it and used it for one side of `sup_eq_closure`.